docs/unpackaged.rst

```diff
@@ -1,14 +1,212 @@
+
+* ``%%%NAMESPACE%%%`` is replaced with the package’s namespace in any context with a namespace (such as a namespaced org or managed org). Otherwise, it remains blank.
+* ``%%%NAMESPACED_ORG%%%`` is replaced with the package’s namespace in a namespaced org *only*, not in a managed installation. Otherwise, it remains blank.
+    .. note:: This token supports use cases where components in one unpackaged metadata bundle refer to components in another, and the dependency bundle acquires a namespace by being deployed into a namespaced org.
+* ``%%%NAMESPACE_OR_C%%%`` is replaced with the package’s namespace in any context with a namespace (such as a namespaced org or managed org). Otherwise, it is replaced with ``c``, the generic namespace used in Lightning components.
+* ``%%%NAMESPACED_ORG_OR_C%%%`` is replaced with the package's namespace in a namespaced org *only*, not in a managed installation. Otherwise, it is replaced with ``c``, the generic namespace used in Lightning components.
+* ``%%%NAMESPACE_DOT%%%`` is replaced with the package’s namespace in any context with a namespace (such as a namespaced org or managed org) followed by a period (``.``) rather than two underscores.
+    .. note:: This token is used to construct references to packaged Record Types.
+
+..
+
+    Example: A portion of metadata from the Nonprofit Success Pack (NPSP) is stored in a subdirectory under ``unpackaged/post``, meaning it's deployed after the application metadata. This metadata updates a Compact Layout on the ``Account`` object, and references packaged metadata from the application as well as from other managed packages. To complete these tasks, this metadata requires the use of namespace tokens to represent the ``npsp`` namespace, letting CumulusCI automatically adapt the metadata to deploy into managed and unmanaged contexts.
+
+.. code-block:: xml
+
+    <?xml version="1.0" encoding="UTF-8"?>
+    <CustomObject xmlns="http://soap.sforce.com/2006/04/metadata">
+        <compactLayouts>
+            <fullName>NPSP_Household_Account</fullName>
+            <fields>Name</fields>
+            <fields>npo02__TotalOppAmount__c</fields>
+            <fields>%%%NAMESPACE%%%Number_of_Household_Members__c</fields>
+            <label>NPSP Household Account</label>
+        </compactLayouts>
+    </CustomObject>
+
+..
+
+    Note that only the reference to the NPSP field ``Number_of_Household_Members__c`` is tokenized. (This field is called ``npsp__Number_of_Household_Members__c`` when installed as part of the managed package.) References to NPSP's own managed package dependency, ``npo02``, are not tokenized because this metadata is always namespaced when installed.
+
+    If this metadata isn't tokenized, it fails to deploy into an org containing NPSP as a beta or released managed package.
+
```

Comment:
Why? This example might need more clarification, or it might need to be swapped out with one easier to comprehend.

(Or it might be completely fine for an advanced dev. Your call.)

Reply:
```suggestion
    If this metadata isn't tokenized, it fails to deploy into an org containing NPSP as a beta or released managed package — because in that context, the field `Number_of_Household_Members__c` is namespaced, as `npsp__ Number_of_Household_Members__c`, and we must refer to it as such.

```

Reply:
We are pretty deep in the weeds here. We should assume that users reading this section are likely a release engineer familiar with packaging basics. But I think we could maybe make this a little more explicit in how the error actually occurs.

@davidmreed - I need you help on how to finish the example. I believe I'm correct up until "the field would be deployed" part. Is the error on deploy, or is this more related with a problem that occurs in the `package.xml` file, or something else?

```suggestion
    If this metadata isn't tokenized, it will fail to deploy into an org containing NPSP as a beta or released managed package. Since all metadata is retrieved in an unmanaged context (i.e. without a namespace prefix) during development, if there is no `%%%NAMESPACE%%%` token present, the field would be deployed without a namespace, and thus result in an error.
```